### PR TITLE
[#4]: Refactor GetDatabase function

### PR DESCRIPTION
**Pull request by kompiangg**
## Overview

Issue id: #4 

Refactoring GetDatabase Function

## Type of change

- [ ] Bug fix
- [x] Enhancement
- [ ] New feature 
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## How Has This Been Tested?
- [ ] New unit tests added.
- [x] Manual tested.

## Checklist:
- [ ] I have commented my code, particularly in hard-to-understand areas.

### CODE REVIEW

File: database.go [hunk 1]
```diff
@@ -17,29 +17,20 @@ func GetDatabase(dbAddress string, dbUsername string, dbPassword string, dbName
...
-		for {
-			log.Println("INFO GetDatabase re-attempting to reconnect to database...")
-			time.Sleep(1 * time.Second)
-			db, err = sql.Open("mysql", dataSourceName)
-			if err == nil {
-				break
+	var db *sql.DB
+	for {
+		db, err := sql.Open("mysql", dataSourceName)
+		if pingErr := db.Ping(); err != nil || pingErr != nil {
```
Comment (audipasuatmadi) [inline database.go:23]: is there a guarantee that db will never be `nil` in case `err` happen in the first place? this could lead to runtime error

Comment (kompiangg) [reply database.go:23]: hmm, I think the previous code isn't check `db` is `nil` or not too

Comment (audipasuatmadi) [reply database.go:23]: approved, kompiang is right. resolved in Discord

Comment (kompiangg) [reply database.go:23]: References:
https://stackoverflow.com/questions/32345124/why-does-sql-open-return-nil-as-error-when-it-should-not

**Review by audipasuatmadi — APPROVED** [review]

**Review by maheswaradevo — APPROVED** [review]
The codes look more convenient and clean. Nice

**Review by nmluci — APPROVED** [review]
nice and clean

**Pull request merged**